File: swift/lib/Sources/Message/Message.swift
public class Message {
    var data: [String: AnyCodable?] = [:]

    public func get<T>(_ name: String) -> T? {
        guard let stored = data[name], let value = stored?.value else {
            return nil
        }

        guard let typed = value as? T else {
            Log.e("[Message : get] Value of \"\(name)\" has another type")
            return nil
        }

        return typed
    }

    public func set(_ name: String, _ value: AnyCodable?) {
        data[name] = value
    }
}

File: kotlin/shared/src/com/xplpc/message/Message.kt
package com.xplpc.message

import com.xplpc.util.Log

class Message {
    @PublishedApi
    internal val values = HashMap<String, Any>()

    fun set(name: String, value: Any) {
        values[name] = value
    }

    inline fun <reified T> get(name: String): T? {
        // The type is reified so the stored value is really checked, since an erased cast would only fail later at the call site.

        val stored = values[name] ?: return null

        if (stored is T) {
            return stored
        }

        Log.e("[Message : get] Value of \"$name\" has another type")

        return null
    }
}
